### PR TITLE
Fix shift break code order for PC XT and VNC drivers

### PR DESCRIPTION
**Pull request by call**
When typing an upper case character, the scancode for the base key is wrapped with the make/break codes for "Left Shift" (`2a/aa`). For example, typing the letter "N" (`31/f0`) would use the scancode `2a 31 f0 aa`.

This PR fixes an issue with the last two scancodes when typing a character that requires shift. Currently, the last two scancodes are inverted, e.g. `2a 31 aa f0`. This results in Left Shift being released _prior_ to the base key's break code.

I noticed this issue while trying to debug the `prltype.py` script used for typing the boot command in the Parallels builder with macOS VMs (another PR to come). Problems caused by this early `aa` scancode seem more prominent on macOS guests—it may be due to worse guest VM performance. For example, I would often see a colon in the boot command typed as `:;`—the base key was still being held as shift was released, resulting in the upper case character followed by the base character.

Note: I am a newbie to Go but gave this my best shot and I welcome any feedback.

### CODE REVIEW

**Comment by mwhooker** [comment]
This is awesome, good catch! 

I'll let someone from the packer team +1 this, but it looks good to me.

Probably the same thing should be done to the `vnc` driver. I tried my hand but had an issue running the tests. Totally okay if you don't want to add that, but wanted to call it out here.

**Review by sylviamoss — APPROVED** [review]
This looks good. 👍🏼  
Could the same problem happen with the KeyOn|KeyPress?

**Comment by call** [comment]
Thanks for the 👍 and comments on this.

@mwhooker -

It looks like the VNC driver implementation of `SendKey()` doesn't suffer from this same bug. The VNC driver contains a [third case](https://github.com/hashicorp/packer-plugin-sdk/blob/main/bootcommand/vnc_driver.go#L118-L127) for `KeyPress` actions, with an explicit `<shift make><base make><base break><shift break>` key event sequence. Compare this with the PC XT implementation which [uses two `if` statements](https://github.com/hashicorp/packer-plugin-sdk/blob/main/bootcommand/pc_xt_driver.go#L149-L163) to handle both `KeyOn`/`KeyOff` and `KeyPress` actions.

I was able to get tests running and added a test for shift functionality to `vnc_driver_test.go`.

@sylviamoss -

> Could the same problem happen with the KeyOn|KeyPress?

I don't think so. `KeyOn|KeyPress` is working properly, always prepending the Left Shift make code (`2a`) to the base key make code, i.e. `2a <base key make code>`. 

The problem is isolated to `KeyOff|KeyPress` because Left Shift's code (`aa`) is _also_ prepended to the base key break code, resulting in a scancode pair of `aa <base key break code>`. This is not correct; Left Shift break should always be last.

File: bootcommand/vnc_driver_test.go [hunk 1]
```diff
@@ -39,6 +39,29 @@ func Test_vncSpecialLookup(t *testing.T) {
 	assert.Equal(t, expected, s.e)
 }
 
+func Test_vncShift(t *testing.T) {
+	in := "AbC"
```
[the remark this answers is not in the document]
Comment (call) [reply bootcommand/vnc_driver_test.go:43]: @mwhooker Good point—an edge case, but might as well handle it. Ok, I updated the implementation and added the specific test you suggested. Please let me know if you'd like to see anything else changed.

[the remark this answers is not in the document]
Comment (mwhooker) [reply bootcommand/vnc_driver_test.go:43]: Thank you so much! I really appreciate you going above and beyond to get the VNC driver, too.

**Review by azr — APPROVED** [review]
Nice one !

**Review by call — COMMENTED** [review]
I had a nitpicky style question about `pc_xt_driver.go` [L50-51](https://github.com/hashicorp/packer-plugin-sdk/blob/main/bootcommand/pc_xt_driver.go#L50-L51). Is there a reason the second URL is indented so much?

```go
	// Scancodes reference: https://www.win.tue.nl/~aeb/linux/kbd/scancodes-1.html
	//						https://www.win.tue.nl/~aeb/linux/kbd/scancodes-10.html
```	

This looks better to my eyes, but i'm not vary familiar with Go idioms:
```go
	// Scancodes reference: https://www.win.tue.nl/~aeb/linux/kbd/scancodes-1.html
	//			https://www.win.tue.nl/~aeb/linux/kbd/scancodes-10.html
```

Figured I'd ask since this PR's still awaiting merge, thanks!

**Comment by mwhooker** [comment]
@call 

> I had a nitpicky style question about pc_xt_driver.go L50-51. Is there a reason the second URL is indented so much?

Good catch! This must be because my editor has a different tabwidth than what github uses. 

on my screen it looks like

Screen Shot 2021-04-07 at 8 55 44 PM

+1 to replacing the tabs with spaces.

**Comment by call** [comment]
@mwhooker 

> +1 to replacing the tabs with spaces.

Done. Tyvm for the quick .02.
